src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java

```diff
@@ -573,6 +595,18 @@ private AnomalyDetectorSettings() {}
      */
     public static int CHECKPOINT_WRITE_QUEUE_SIZE_IN_BYTES = 200_000;
 
+    /**
+     * CheckpointMaintainRequest has model Id (roughly 256 bytes), and QueuedRequest
+     * fields including detector Id(roughly 128 bytes), expirationEpochMs (long,
+     *  8 bytes), and priority (12 bytes).
+     * Plus Java object size (12 bytes), we have roughly 416 bytes per request.
+     * We don't want the total size exceeds 0.1% of the heap.
+     * We can have at most 0.1% heap / 416 = heap / 416,000.
+     * For t3.small, 0.1% heap is of 1MB. The queue's size is up to
+     * 10^ 6 / 416 = 2403
+     */
+    public static int CHECKPOINT_MAINTAIN_REQUEST_SIZE_IN_BYTES = 416;
```

Comment:
So given a single t3.small instance, we can optimally support 2403 entities per interval right? So given a frequency of 6 hrs, then given a detector with ~14k models would be able to be entirely saved. 

Is there any guidance from product or community on the tradeoff regarding the frequency of saving models / the importance of saving all models vs. how often they are saved?

Reply:
It is a tradeoff: the more frequent, the more refreshed the saved model, but more resource consuming. I can add that as new doc in 2.2.

Reply:
The request size is one aspect. Together with queue size, they determine how many requests we can queue. But how many checkpoints we can really save depends on a few more factors (e.g., cluster load, the number of models in the cluster).